docs/notebooks/Poincare Evaluation.ipynb

```diff
@@ -0,0 +1,1553 @@
+    "This is the list of tasks - \n",
+    "1. WordNet reconstruction\n",
+    "2. WordNet link prediction\n",
+    "3. Link prediction in collaboration networks (evaluation incomplete)\n",
+    "4. Lexical entailment on HyperLex\n",
+    "\n",
+    "A more detailed explanation of the tasks and the evaluation methodology is present in the individual evaluation subsections."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 1. Setup\n",
+    "\n",
+    "The following code clones the `poincare-embedding` repository containing the C++ implementation of the Poincare embeddings, and applies a patch containing minor additions to the implementation. Please set the variable `parent_directory` below to define the directory to which the repository is cloned."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "current_directory = os.getcwd()\n",
+    "\n",
+    "# Change this variable to `False` to not remove and re-download repos for external implementations\n",
+    "force_setup = True\n",
+    "\n",
+    "# The poincare datasets, models and source code for external models are downloaded to this directory\n",
+    "parent_directory = os.path.join(current_directory, 'poincare')\n",
+    "! mkdir -p {parent_directory}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 111,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "/home/jayant/projects/gensim/docs/notebooks/poincare\n",
+      "Cloning into 'poincare-np-embedding'...\n",
+      "remote: Counting objects: 20, done.\u001b[K\n",
+      "remote: Compressing objects: 100% (18/18), done.\u001b[K\n",
+      "remote: Total 20 (delta 2), reused 20 (delta 2), pack-reused 0\u001b[K\n",
+      "Unpacking objects: 100% (20/20), done.\n",
+      "Checking connectivity... done.\n",
+      "Cloning into 'poincare-cpp-embedding'...\n",
+      "remote: Counting objects: 96, done.\u001b[K\n",
+      "remote: Total 96 (delta 0), reused 0 (delta 0), pack-reused 96\u001b[K\n",
+      "Unpacking objects: 100% (96/96), done.\n",
+      "Checking connectivity... done.\n"
+     ]
+    }
+   ],
+   "source": [
+    "% cd {parent_directory}\n",
+    "\n",
+    "# Clone repos\n",
+    "np_repo_name = 'poincare-np-embedding'\n",
+    "if force_setup and os.path.exists(np_repo_name):\n",
+    "    ! rm -rf {np_repo_name}\n",
+    "if not os.path.exists(np_repo_name):\n",
+    "    ! git clone https://github.com/nishnik/poincare_embeddings.git {np_repo_name}\n",
+    "\n",
+    "cpp_repo_name = 'poincare-cpp-embedding'\n",
+    "if force_setup and os.path.exists(cpp_repo_name):\n",
+    "    ! rm -rf {cpp_repo_name}\n",
+    "if not os.path.exists(cpp_repo_name):\n",
+    "    ! git clone https://github.com/TatsuyaShirakawa/poincare-embedding.git {cpp_repo_name}\n",
+    "\n",
+    "patches_applied = False"
```

Comment:
This looks wrong. If this is re-run with `force_setup == False` the patches will get applied twice.

Reply:
Fixed